pipeline/metadata/hyperquack_outcome.py

```diff
@@ -0,0 +1,170 @@
+  if "failed to parse Location header" in error: return "http/http.invalid"
+  if "malformed HTTP" in error: return "http/http.invalid"
+  if "malformed MIME" in error: return "http/http.invalid"
+
+  # Content verification failures
+  if scan_type != "echo" and error.endswith("EOF"): return "content/mismatch" # Echo
+  # Hyperquack v1 errors
+  if error == "Incorrect echo response": return "content/mismatch" # Echo
+  if error == "Received response": return "content/mismatch" # Discard
+  if error == "Incorrect web response: status lines don't match": return _status_mismatch(received_status) # HTTP/S
+  if error == "Incorrect web response: bodies don't match": return "content/body_mismatch" # HTTP/S
+  if error == "Incorrect web response: certificates don't match": return "content/tls_mismatch" # HTTPS
+  if error == "Incorrect web response: cipher suites don't match": return "content/tls_mismatch" # HTTPS
+  if error == "Incorrect web response: TLS versions don't match": return "content/tls_mismatch" # HTTPS
+  # Hyperquack v2 errors
+  if error == "echo response does not match echo request": return "content/mismatch" # Echo
+  if error == "discard response is not empty": return "content/mismatch" # Discard
+  if error == "Status lines does not match": return _status_mismatch(received_status) # HTTP/S
+  if error == "Bodies do not match": return "content/body_mismatch" # HTTPS
+  if error == "Certificates do not match": return "content/tls_mismatch" # HTTPS
+  if error == "Cipher suites do not match": return "content/tls_mismatch" # HTTPS
+  if error == "TLS versions do not match": return "content/tls_mismatch" # HTTPS
+  # Hyperquack v2 multiple errors
+  if error == "Status lines do not match": return _status_mismatch(received_status) # HTTP/S
+  if error == "Bodies do not match": return "content/body_mismatch" # HTTP/S
+  if "header field missing" in error: return "content/header_mismatch" # HTTP/S
+  if "header field mismatch" in error: return "content/header_mismatch" # HTTP/S
+
+  return "unknown/unknown"
+  # yapf: enable
+
+
+def classify_hyperquack_outcome(error: Optional[str], scan_type: str,
+                                received_status: Optional[str],
+                                matches_template: Optional[bool],
+                                is_known_blockpage: Optional[bool],
+                                blockpage_fingerprint: Optional[str],
+                                received_headers: List[str]) -> str:
+  """
+  Args:
+    error: string like "Incorrect web response: status lines don't match"
+    received_status: HTTP status like "200 OK"
+    matches_template: did the response match the control template
+    is_known_blockpage: did the page content match a blockpage
+    blockpage_fingerprint: which known blockpage did we match
+    received_headers: HTTP headers
+
+
+  Returns:
+    outcome string like "read/tcp.reset"
+  """
+  if is_known_blockpage and blockpage_fingerprint:
```

Comment:
Maybe add a TODO to order this by when we see the signals.

Reply:
done